backend/game/deck/deck_handler.py

```diff
@@ -94,6 +95,7 @@ def buy_card_from_store(self, index, purchasing_player: Player, other_players):
     # optional second arg allows discard to handle removing card from origin as well
     def discard(self, card: Card, card_from_location: List[Card] = None):
         self.__discard_pile.append(card)
+        self.cards_swept.append(card)
```

Comment:
I don't think this is where you want to get the card swept info. There's a method specifically to sweep the card store. Sweeping gets rid of all cards. I think this discard method  is called any time a discard card is purchased. This would give you the wrong data. You probably want to handle the DB stuff at the Consumers.py level anyways, then whenever the card sweep request is called, just grab the current state of card store using `cards_to_sweep = state.deckhandler.card_store` or whatever it is right before actually sweeping

Reply:
Thanks for those details they were very helpful, took a different approach thou in that i capture the top 3 card in deck_handler and then just pass over to consumers.py, and the solution works as needed